clang/test/CodeGen/fake-use-small-aggs.c

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 %s -O2 -emit-llvm -fextend-lifetimes -disable-llvm-passes -o - | FileCheck %s
+// Check that we generate a fake_use call for small aggregate types.
+
+// CHECK-DAG:  %[[FAKEUSE1:[^ ]+]] = load{{.*}} %loc,
+// CHECK-DAG:  call{{.*}}llvm.fake.use({{.*}}%[[FAKEUSE1]])
+// CHECK-DAG:  %[[FAKEUSE2:[^ ]+]] = load{{.*}} %arr,
+// CHECK-DAG:  call{{.*}}llvm.fake.use({{.*}}%[[FAKEUSE2]])
+// CHECK-DAG:  %[[FAKEUSE3:[^ ]+]] = load{{.*}} %S,
+// CHECK-DAG:  call{{.*}}llvm.fake.use({{.*}}%[[FAKEUSE3]])
```

Comment:
Better to remove the -DAG part and explicitly test that the fake uses are at the end of the function?